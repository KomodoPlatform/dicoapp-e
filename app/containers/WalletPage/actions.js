// @flow

import {
  LOAD_TRANSACTIONS,
  LOAD_TRANSACTION_SUCCESS,
  LOAD_TRANSACTIONS_SUCCESS,
  LOAD_TRANSACTIONS_ERROR
} from './constants';
import type { TransactionPayload } from './schema';

export function loadTransactions() {
  return {
    type: LOAD_TRANSACTIONS
  };
}

export function loadTransactionsSuccess() {
  return {
    type: LOAD_TRANSACTIONS_SUCCESS
  };
}

export function loadTransactionSuccess(transaction: Array<TransactionPayload>) {
  return {
    type: LOAD_TRANSACTION_SUCCESS,
    payload: {
      transaction
    }
  };
}

export function loadTransactionsError(message: string) {
  return {
    type: LOAD_TRANSACTIONS_ERROR,
    error: {
      message
    }
  };
}
