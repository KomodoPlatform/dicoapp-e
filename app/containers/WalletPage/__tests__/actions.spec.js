import { loadTransactions, loadTransactionsSuccess } from '../actions';
import { LOAD_TRANSACTIONS, LOAD_TRANSACTIONS_SUCCESS } from '../constants';

describe('containers/WalletPage/actions/loadTransactions', () => {
  it('should loadTransactions should create loadTransactions action', () => {
    expect(loadTransactions()).toMatchSnapshot();
  });

  it('should return the correct type and the passed name', () => {
    const expectedResult = {
      type: LOAD_TRANSACTIONS
    };

    expect(loadTransactions()).toEqual(expectedResult);
  });
});

describe('containers/WalletPage/actions/loadTransactionsSuccess', () => {
  it('should loadTransactionsSuccess should create loadTransactionsSuccess action', () => {
    expect(loadTransactionsSuccess()).toMatchSnapshot();
  });

  it('should return the correct type and the passed name', () => {
    const expectedResult = {
      type: LOAD_TRANSACTIONS_SUCCESS
    };

    expect(loadTransactionsSuccess()).toEqual(expectedResult);
  });
});
