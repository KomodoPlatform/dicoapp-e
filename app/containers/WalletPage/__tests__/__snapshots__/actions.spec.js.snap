// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletPage Actions loadTransactions should loadTransactions should create loadTransactions action 1`] = `
Object {
  "type": "dicoapp/WalletPage/LOAD_TRANSACTIONS",
}
`;

exports[`containers/WalletPage/actions/loadTransactions should loadTransactions should create loadTransactions action 1`] = `
Object {
  "type": "dicoapp/WalletPage/LOAD_TRANSACTIONS",
}
`;

exports[`containers/WalletPage/actions/loadTransactionsSuccess should loadTransactionsSuccess should create loadTransactionsSuccess action 1`] = `
Object {
  "type": "dicoapp/WalletPage/LOAD_TRANSACTIONS_SUCCESS",
}
`;
